pip/req/req_set.py

```diff
@@ -123,6 +133,68 @@ def prep_for_dist(self):
         self.req_to_install.assert_source_matches_version()
 
 
+class SetupCfgDistribution(pkg_resources.Distribution):
+
+    def __init__(self, cfg, location=None):
+        self._cfg = cfg
+        project_name = cfg.get('metadata', 'name')
+        super(SetupCfgDistribution, self).__init__(
+            location=location, project_name=project_name)
+
+    @property
+    def _dep_map(self):
```

Comment:
I'm going to guess that most people aren't familiar enough with `pkg_resources.Distribution` to understand why this would be necessary. Would you mind adding a comment/docstring explaining it for future developers?